noblocknoparty/app
# Scan QRcode

## Changes

- Move search to use local state to global state
- Add QRCode mutation and query
- Display the first 4~5 ETH address if you are the contract admin
- Shows "Scan QR code" only via supported browser
- Scans QRcode and search by the address

## Demo video

https://www.youtube.com/watch?v=9IRkOWOvC5k&feature=youtu.be

(Tested on Cipher and Status.im, Coinbase Wallet is not supported)

## How to test

1. Change `src/config/env.json` to point to rinkeby

```
{
  "DEPLOYER_CONTRACT_ADDRESS": "0xfe4d296f7b7e5c0065d5df7865075d93d88415e1",
  "API_URL": "https://rinkeby.api.kickback.events",
  "NUM_CONFIRMATIONS": 1
}
```

2. Install https://ngrok.com and proxy your local node

```
ngrok http 3000
```

NOTE: Do not hit the URL end point too often as it has fairly low rate limiting 

3. Open the proxy url from mobile wallet (Status, Ciphyer, Coinbase wallet)

4. Create test event (or create it via metamask using the same seed as the one you have on your mobile wallet)

5. RSVP from an account

6. Open the RSVPed account via Etherscan (which has QR code section)

7. Click "Scan QR code" and scan the QRCode of #6.

8. It should fill in the scanned QR code address into search box.
